File: badapple.js
function trackChord () {
    for (let index = 0; index < 2; index++) {
        loopMelody2()
        music.playTone(156, music.beat(BeatFraction.Half))
        music.playTone(185, music.beat(BeatFraction.Quarter))
        music.playTone(156, music.beat(BeatFraction.Quarter))
        music.playTone(208, music.beat(BeatFraction.Half))
        music.playTone(185, music.beat(BeatFraction.Quarter))
        music.playTone(208, music.beat(BeatFraction.Quarter))
        loopMelody2()
        music.playTone(208, music.beat(BeatFraction.Half))
        music.playTone(185, music.beat(BeatFraction.Quarter))
        music.playTone(208, music.beat(BeatFraction.Quarter))
        music.playTone(185, music.beat(BeatFraction.Half))
        music.playTone(156, music.beat(BeatFraction.Quarter))
        music.playTone(185, music.beat(BeatFraction.Quarter))
    }
    for (let index = 0; index < 3; index++) {
        loopMelody3()
    }
    loopMelody2()
    music.playTone(156, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    music.playTone(156, music.beat(BeatFraction.Quarter))
    music.playTone(208, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    music.playTone(208, music.beat(BeatFraction.Quarter))
    for (let index = 0; index < 2; index++) {
        // 낮은 시
        music.playTone(123.47, music.beat(BeatFraction.Half))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        music.rest(music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        // 낮은 라#
        music.playTone(116.54, music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
    }
    music.playTone(139, music.beat(BeatFraction.Half))
    music.playTone(139, music.beat(BeatFraction.Half))
    music.playTone(156, music.beat(BeatFraction.Half))
    music.playTone(156, music.beat(BeatFraction.Half))
    music.playTone(175, music.beat(BeatFraction.Half))
    music.playTone(175, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Half))
    for (let index = 0; index < 3; index++) {
        loopMelody4()
    }
    for (let index = 0; index < 2; index++) {
        // 낮은 시
        music.playTone(123.47, music.beat(BeatFraction.Half))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        music.rest(music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        // 낮은 라#
        music.playTone(116.54, music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
    }
    music.playTone(139, music.beat(BeatFraction.Half))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.rest(music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(123.47, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Half))
    music.playTone(165, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(185, music.beat(BeatFraction.Half))
    music.playTone(165, music.beat(BeatFraction.Quarter))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    loopMelody2()
    music.playTone(208, music.beat(BeatFraction.Whole))
    music.playTone(185, music.beat(BeatFraction.Whole))
    playChord = false
}
function sndDrum (duration: number, repeat: number) {
    for (let index = 0; index < repeat; index++) {
        music.playTone(100, 10)
        music.rest(duration)
    }
}
// Chord용 반복 1
function loopMelody2 () {
    for (let index = 0; index < 3; index++) {
        music.playTone(156, music.beat(BeatFraction.Half))
        music.playTone(156, music.beat(BeatFraction.Quarter))
        music.playTone(156, music.beat(BeatFraction.Quarter))
        music.rest(music.beat(BeatFraction.Quarter))
        music.playTone(156, music.beat(BeatFraction.Quarter))
        music.playTone(139, music.beat(BeatFraction.Quarter))
        music.playTone(156, music.beat(BeatFraction.Quarter))
    }
}
input.onButtonPressed(Button.A, function () {
    isPlaying = 1
    playBadApple()
})
function loopMelody1 () {
    music.playTone(622, music.beat(BeatFraction.Half))
    music.playTone(698, music.beat(BeatFraction.Half))
    music.playTone(740, music.beat(BeatFraction.Half))
    music.playTone(831, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Whole))
    music.playTone(1244, music.beat(BeatFraction.Half))
    music.playTone(1108, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Whole))
    music.playTone(622, music.beat(BeatFraction.Whole))
    music.playTone(932, music.beat(BeatFraction.Half))
    music.playTone(831, music.beat(BeatFraction.Half))
    music.playTone(740, music.beat(BeatFraction.Half))
    music.playTone(698, music.beat(BeatFraction.Half))
    music.playTone(622, music.beat(BeatFraction.Half))
    music.playTone(698, music.beat(BeatFraction.Half))
    music.playTone(740, music.beat(BeatFraction.Half))
    music.playTone(831, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Whole))
    music.playTone(831, music.beat(BeatFraction.Half))
    music.playTone(740, music.beat(BeatFraction.Half))
}
function trackBeat () {
    for (let index = 0; index < 2; index++) {
        for (let index = 0; index < 3; index++) {
            sndDrum(music.beat(BeatFraction.Whole), 3)
            sndDrum(music.beat(BeatFraction.Quarter), 4)
            sndDrum(music.beat(BeatFraction.Whole), 3)
            sndDrum(music.beat(BeatFraction.Half), 2)
        }
        sndDrum(music.beat(BeatFraction.Whole), 3)
        sndDrum(music.beat(BeatFraction.Quarter), 4)
        sndDrum(music.beat(BeatFraction.Whole), 4)
    }
    playBeat = false
}
// Chord용 반복 2
function loopMelody3 () {
    loopMelody2()
    music.playTone(156, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    music.playTone(156, music.beat(BeatFraction.Quarter))
    music.playTone(208, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    music.playTone(208, music.beat(BeatFraction.Quarter))
    for (let index = 0; index < 2; index++) {
        // 낮은 시
        music.playTone(123.47, music.beat(BeatFraction.Half))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        music.rest(music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        // 낮은 라#
        music.playTone(116.54, music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
    }
    music.playTone(139, music.beat(BeatFraction.Half))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.rest(music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(123.47, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(156, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    music.playTone(156, music.beat(BeatFraction.Quarter))
    music.playTone(208, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    music.playTone(208, music.beat(BeatFraction.Quarter))
}
// Chord용 반복 3
function loopMelody4 () {
    for (let index = 0; index < 2; index++) {
        // 낮은 시
        music.playTone(123.47, music.beat(BeatFraction.Half))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        music.rest(music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
        // 낮은 라#
        music.playTone(116.54, music.beat(BeatFraction.Quarter))
        music.playTone(123.47, music.beat(BeatFraction.Quarter))
    }
    music.playTone(139, music.beat(BeatFraction.Half))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.rest(music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(123.47, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Half))
    music.playTone(165, music.beat(BeatFraction.Quarter))
    music.playTone(139, music.beat(BeatFraction.Quarter))
    music.playTone(185, music.beat(BeatFraction.Half))
    music.playTone(165, music.beat(BeatFraction.Quarter))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    loopMelody2()
    music.playTone(156, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    music.playTone(156, music.beat(BeatFraction.Quarter))
    music.playTone(208, music.beat(BeatFraction.Half))
    music.playTone(185, music.beat(BeatFraction.Quarter))
    music.playTone(208, music.beat(BeatFraction.Quarter))
}
input.onButtonPressed(Button.AB, function () {
    trackChord()
})
function trackMain () {
    for (let index = 0; index < 2; index++) {
        loopMelody1()
        music.playTone(698, music.beat(BeatFraction.Half))
        music.playTone(622, music.beat(BeatFraction.Half))
        music.playTone(698, music.beat(BeatFraction.Half))
        music.playTone(740, music.beat(BeatFraction.Half))
        music.playTone(698, music.beat(BeatFraction.Half))
        music.playTone(622, music.beat(BeatFraction.Half))
        music.playTone(587, music.beat(BeatFraction.Half))
        music.playTone(698, music.beat(BeatFraction.Half))
        loopMelody1()
        music.playTone(698, music.beat(BeatFraction.Whole))
        music.playTone(740, music.beat(BeatFraction.Whole))
        music.playTone(831, music.beat(BeatFraction.Whole))
        music.playTone(932, music.beat(BeatFraction.Whole))
    }
    for (let index = 0; index < 3; index++) {
        for (let index = 0; index < 2; index++) {
            music.playTone(1108, music.beat(BeatFraction.Half))
            music.playTone(1244, music.beat(BeatFraction.Half))
            music.playTone(932, music.beat(BeatFraction.Half))
            music.playTone(831, music.beat(BeatFraction.Half))
            music.playTone(932, music.beat(BeatFraction.Whole))
            music.playTone(831, music.beat(BeatFraction.Half))
            music.playTone(932, music.beat(BeatFraction.Half))
        }
        music.playTone(831, music.beat(BeatFraction.Half))
        music.playTone(740, music.beat(BeatFraction.Half))
        music.playTone(698, music.beat(BeatFraction.Half))
        music.playTone(554, music.beat(BeatFraction.Half))
        music.playTone(622, music.beat(BeatFraction.Whole))
        music.playTone(554, music.beat(BeatFraction.Half))
        music.playTone(622, music.beat(BeatFraction.Half))
        music.playTone(698, music.beat(BeatFraction.Half))
        music.playTone(740, music.beat(BeatFraction.Half))
        music.playTone(831, music.beat(BeatFraction.Half))
        music.playTone(932, music.beat(BeatFraction.Half))
        music.playTone(622, music.beat(BeatFraction.Whole))
        music.playTone(831, music.beat(BeatFraction.Half))
        music.playTone(932, music.beat(BeatFraction.Half))
    }
    music.playTone(1108, music.beat(BeatFraction.Half))
    music.playTone(1244, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Half))
    music.playTone(831, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Whole))
    music.playTone(831, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Half))
    music.playTone(1108, music.beat(BeatFraction.Half))
    music.playTone(1244, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Half))
    music.playTone(831, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Whole))
    music.playTone(1244, music.beat(BeatFraction.Half))
    music.playTone(1396, music.beat(BeatFraction.Half))
    music.playTone(1479, music.beat(BeatFraction.Half))
    music.playTone(1396, music.beat(BeatFraction.Half))
    music.playTone(1244, music.beat(BeatFraction.Half))
    music.playTone(1108, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Whole))
    music.playTone(831, music.beat(BeatFraction.Half))
    music.playTone(932, music.beat(BeatFraction.Half))
    music.playTone(831, music.beat(BeatFraction.Half))
    music.playTone(740, music.beat(BeatFraction.Half))
    music.playTone(698, music.beat(BeatFraction.Half))
    music.playTone(554, music.beat(BeatFraction.Half))
    music.playTone(622, music.beat(BeatFraction.Whole))
}
input.onButtonPressed(Button.B, function () {
    if (isPlaying == 1) {
        isPlaying = 2
        while (isPlaying == 2) {
            music.stopMelody(MelodyStopOptions.All)
            basic.clearScreen()
        }
    }
    if (isPlaying == 2) {
        isPlaying = 1
        music.setVolume(150)
        while (isPlaying == 1) {
            basic.showString("BADAPPLE")
        }
    }
})
function playBadApple () {
    playBeat = true
    for (let index = 0; index < 8; index++) {
        music.rest(music.beat(BeatFraction.Breve))
    }
    playChord = true
    for (let index = 0; index < 8; index++) {
        music.rest(music.beat(BeatFraction.Breve))
    }
    trackMain()
}
let playChord = false
let playBeat = false
let isPlaying = 0
music.setTempo(138)
music.setBuiltInSpeakerEnabled(true)
music.setVolume(150)
isPlaying = 0
playBeat = false
playChord = false
basic.showIcon(IconNames.Yes)
basic.pause(500)
basic.clearScreen()
basic.forever(function () {
    while (isPlaying == 1) {
        basic.showString("BADAPPLE")
    }
})
basic.forever(function () {
    while (playBeat == true) {
        trackBeat()
    }
})
basic.forever(function () {
    while (playChord == true) {
        trackChord()
    }
})

File: badapple.py
def trackChord():
    global playChord
    for index in range(2):
        loopMelody2()
        music.play_tone(156, music.beat(BeatFraction.HALF))
        music.play_tone(185, music.beat(BeatFraction.QUARTER))
        music.play_tone(156, music.beat(BeatFraction.QUARTER))
        music.play_tone(208, music.beat(BeatFraction.HALF))
        music.play_tone(185, music.beat(BeatFraction.QUARTER))
        music.play_tone(208, music.beat(BeatFraction.QUARTER))
        loopMelody2()
        music.play_tone(208, music.beat(BeatFraction.HALF))
        music.play_tone(185, music.beat(BeatFraction.QUARTER))
        music.play_tone(208, music.beat(BeatFraction.QUARTER))
        music.play_tone(185, music.beat(BeatFraction.HALF))
        music.play_tone(156, music.beat(BeatFraction.QUARTER))
        music.play_tone(185, music.beat(BeatFraction.QUARTER))
    for index2 in range(3):
        loopMelody3()
    loopMelody2()
    music.play_tone(156, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    music.play_tone(156, music.beat(BeatFraction.QUARTER))
    music.play_tone(208, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    music.play_tone(208, music.beat(BeatFraction.QUARTER))
    for index3 in range(2):
        # 낮은 시
        music.play_tone(123.47, music.beat(BeatFraction.HALF))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        music.rest(music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        # 낮은 라#
        music.play_tone(116.54, music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.HALF))
    music.play_tone(139, music.beat(BeatFraction.HALF))
    music.play_tone(156, music.beat(BeatFraction.HALF))
    music.play_tone(156, music.beat(BeatFraction.HALF))
    music.play_tone(175, music.beat(BeatFraction.HALF))
    music.play_tone(175, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.HALF))
    for index4 in range(3):
        loopMelody4()
    for index5 in range(2):
        # 낮은 시
        music.play_tone(123.47, music.beat(BeatFraction.HALF))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        music.rest(music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        # 낮은 라#
        music.play_tone(116.54, music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.HALF))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.rest(music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.HALF))
    music.play_tone(165, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(185, music.beat(BeatFraction.HALF))
    music.play_tone(165, music.beat(BeatFraction.QUARTER))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    loopMelody2()
    music.play_tone(208, music.beat(BeatFraction.WHOLE))
    music.play_tone(185, music.beat(BeatFraction.WHOLE))
    playChord = False
def sndDrum(duration: number, repeat: number):
    for index6 in range(repeat):
        music.play_tone(100, 10)
        music.rest(duration)
# Chord용 반복 1
def loopMelody2():
    for index7 in range(3):
        music.play_tone(156, music.beat(BeatFraction.HALF))
        music.play_tone(156, music.beat(BeatFraction.QUARTER))
        music.play_tone(156, music.beat(BeatFraction.QUARTER))
        music.rest(music.beat(BeatFraction.QUARTER))
        music.play_tone(156, music.beat(BeatFraction.QUARTER))
        music.play_tone(139, music.beat(BeatFraction.QUARTER))
        music.play_tone(156, music.beat(BeatFraction.QUARTER))

def on_button_pressed_a():
    global isPlaying
    isPlaying = 1
    playBadApple()
input.on_button_pressed(Button.A, on_button_pressed_a)

def loopMelody1():
    music.play_tone(622, music.beat(BeatFraction.HALF))
    music.play_tone(698, music.beat(BeatFraction.HALF))
    music.play_tone(740, music.beat(BeatFraction.HALF))
    music.play_tone(831, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.WHOLE))
    music.play_tone(1244, music.beat(BeatFraction.HALF))
    music.play_tone(1108, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.WHOLE))
    music.play_tone(622, music.beat(BeatFraction.WHOLE))
    music.play_tone(932, music.beat(BeatFraction.HALF))
    music.play_tone(831, music.beat(BeatFraction.HALF))
    music.play_tone(740, music.beat(BeatFraction.HALF))
    music.play_tone(698, music.beat(BeatFraction.HALF))
    music.play_tone(622, music.beat(BeatFraction.HALF))
    music.play_tone(698, music.beat(BeatFraction.HALF))
    music.play_tone(740, music.beat(BeatFraction.HALF))
    music.play_tone(831, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.WHOLE))
    music.play_tone(831, music.beat(BeatFraction.HALF))
    music.play_tone(740, music.beat(BeatFraction.HALF))
def trackBeat():
    global playBeat
    for index8 in range(2):
        for index9 in range(3):
            sndDrum(music.beat(BeatFraction.WHOLE), 3)
            sndDrum(music.beat(BeatFraction.QUARTER), 4)
            sndDrum(music.beat(BeatFraction.WHOLE), 3)
            sndDrum(music.beat(BeatFraction.HALF), 2)
        sndDrum(music.beat(BeatFraction.WHOLE), 3)
        sndDrum(music.beat(BeatFraction.QUARTER), 4)
        sndDrum(music.beat(BeatFraction.WHOLE), 4)
    playBeat = False
# Chord용 반복 2
def loopMelody3():
    loopMelody2()
    music.play_tone(156, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    music.play_tone(156, music.beat(BeatFraction.QUARTER))
    music.play_tone(208, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    music.play_tone(208, music.beat(BeatFraction.QUARTER))
    for index10 in range(2):
        # 낮은 시
        music.play_tone(123.47, music.beat(BeatFraction.HALF))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        music.rest(music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        # 낮은 라#
        music.play_tone(116.54, music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.HALF))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.rest(music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(156, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    music.play_tone(156, music.beat(BeatFraction.QUARTER))
    music.play_tone(208, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    music.play_tone(208, music.beat(BeatFraction.QUARTER))
# Chord용 반복 3
def loopMelody4():
    for index11 in range(2):
        # 낮은 시
        music.play_tone(123.47, music.beat(BeatFraction.HALF))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        music.rest(music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
        # 낮은 라#
        music.play_tone(116.54, music.beat(BeatFraction.QUARTER))
        music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.HALF))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.rest(music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(123.47, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.HALF))
    music.play_tone(165, music.beat(BeatFraction.QUARTER))
    music.play_tone(139, music.beat(BeatFraction.QUARTER))
    music.play_tone(185, music.beat(BeatFraction.HALF))
    music.play_tone(165, music.beat(BeatFraction.QUARTER))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    loopMelody2()
    music.play_tone(156, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    music.play_tone(156, music.beat(BeatFraction.QUARTER))
    music.play_tone(208, music.beat(BeatFraction.HALF))
    music.play_tone(185, music.beat(BeatFraction.QUARTER))
    music.play_tone(208, music.beat(BeatFraction.QUARTER))

def on_button_pressed_ab():
    trackChord()
input.on_button_pressed(Button.AB, on_button_pressed_ab)

def trackMain():
    for index12 in range(2):
        loopMelody1()
        music.play_tone(698, music.beat(BeatFraction.HALF))
        music.play_tone(622, music.beat(BeatFraction.HALF))
        music.play_tone(698, music.beat(BeatFraction.HALF))
        music.play_tone(740, music.beat(BeatFraction.HALF))
        music.play_tone(698, music.beat(BeatFraction.HALF))
        music.play_tone(622, music.beat(BeatFraction.HALF))
        music.play_tone(587, music.beat(BeatFraction.HALF))
        music.play_tone(698, music.beat(BeatFraction.HALF))
        loopMelody1()
        music.play_tone(698, music.beat(BeatFraction.WHOLE))
        music.play_tone(740, music.beat(BeatFraction.WHOLE))
        music.play_tone(831, music.beat(BeatFraction.WHOLE))
        music.play_tone(932, music.beat(BeatFraction.WHOLE))
    for index13 in range(3):
        for index14 in range(2):
            music.play_tone(1108, music.beat(BeatFraction.HALF))
            music.play_tone(1244, music.beat(BeatFraction.HALF))
            music.play_tone(932, music.beat(BeatFraction.HALF))
            music.play_tone(831, music.beat(BeatFraction.HALF))
            music.play_tone(932, music.beat(BeatFraction.WHOLE))
            music.play_tone(831, music.beat(BeatFraction.HALF))
            music.play_tone(932, music.beat(BeatFraction.HALF))
        music.play_tone(831, music.beat(BeatFraction.HALF))
        music.play_tone(740, music.beat(BeatFraction.HALF))
        music.play_tone(698, music.beat(BeatFraction.HALF))
        music.play_tone(554, music.beat(BeatFraction.HALF))
        music.play_tone(622, music.beat(BeatFraction.WHOLE))
        music.play_tone(554, music.beat(BeatFraction.HALF))
        music.play_tone(622, music.beat(BeatFraction.HALF))
        music.play_tone(698, music.beat(BeatFraction.HALF))
        music.play_tone(740, music.beat(BeatFraction.HALF))
        music.play_tone(831, music.beat(BeatFraction.HALF))
        music.play_tone(932, music.beat(BeatFraction.HALF))
        music.play_tone(622, music.beat(BeatFraction.WHOLE))
        music.play_tone(831, music.beat(BeatFraction.HALF))
        music.play_tone(932, music.beat(BeatFraction.HALF))
    music.play_tone(1108, music.beat(BeatFraction.HALF))
    music.play_tone(1244, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.HALF))
    music.play_tone(831, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.WHOLE))
    music.play_tone(831, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.HALF))
    music.play_tone(1108, music.beat(BeatFraction.HALF))
    music.play_tone(1244, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.HALF))
    music.play_tone(831, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.WHOLE))
    music.play_tone(1244, music.beat(BeatFraction.HALF))
    music.play_tone(1396, music.beat(BeatFraction.HALF))
    music.play_tone(1479, music.beat(BeatFraction.HALF))
    music.play_tone(1396, music.beat(BeatFraction.HALF))
    music.play_tone(1244, music.beat(BeatFraction.HALF))
    music.play_tone(1108, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.WHOLE))
    music.play_tone(831, music.beat(BeatFraction.HALF))
    music.play_tone(932, music.beat(BeatFraction.HALF))
    music.play_tone(831, music.beat(BeatFraction.HALF))
    music.play_tone(740, music.beat(BeatFraction.HALF))
    music.play_tone(698, music.beat(BeatFraction.HALF))
    music.play_tone(554, music.beat(BeatFraction.HALF))
    music.play_tone(622, music.beat(BeatFraction.WHOLE))

def on_button_pressed_b():
    global isPlaying
    if isPlaying == 1:
        isPlaying = 2
        while isPlaying == 2:
            music.stop_melody(MelodyStopOptions.ALL)
            basic.clear_screen()
    if isPlaying == 2:
        isPlaying = 1
        music.set_volume(150)
        while isPlaying == 1:
            basic.show_string("BADAPPLE")
input.on_button_pressed(Button.B, on_button_pressed_b)

def playBadApple():
    global playBeat, playChord
    playBeat = True
    for index15 in range(8):
        music.rest(music.beat(BeatFraction.BREVE))
    playChord = True
    for index16 in range(8):
        music.rest(music.beat(BeatFraction.BREVE))
    trackMain()
playChord = False
playBeat = False
isPlaying = 0
music.set_tempo(138)
music.set_built_in_speaker_enabled(True)
music.set_volume(150)
isPlaying = 0
playBeat = False
playChord = False
basic.show_icon(IconNames.YES)
basic.pause(500)
basic.clear_screen()

def on_forever():
    while isPlaying == 1:
        basic.show_string("BADAPPLE")
basic.forever(on_forever)

def on_forever2():
    while playBeat == True:
        trackBeat()
basic.forever(on_forever2)

def on_forever3():
    while playChord == True:
        trackChord()
basic.forever(on_forever3)
